cluster/ec2_pke.go

```diff
@@ -432,93 +455,99 @@ func (c *EC2ClusterPKE) UpdatePKECluster(ctx context.Context, request *pkgCluste
 		return errors.New(fmt.Sprintf("Subnet IDs not found (%v %T)", c.model.Network.CloudProviderConfig["subnets"], c.model.Network.CloudProviderConfig["subnets"]))
 	}
 
-	newNodepools := createNodePoolsFromPKERequest(request.PKE.NodePools)
-	input := pkeworkflow.UpdateClusterWorkflowInput{
-		ClusterID:                   uint(c.GetID()),
-		NodePools:                   newNodepools,
-		OrganizationID:              uint(c.GetOrganizationId()),
-		ClusterUID:                  c.GetUID(),
-		ClusterName:                 c.GetName(),
-		SecretID:                    string(c.GetSecretId()),
-		Region:                      c.GetLocation(),
-		PipelineExternalURL:         externalBaseURL,
-		PipelineExternalURLInsecure: externalBaseURLInsecure,
-		VPCID:                       vpcid,
-		SubnetIDs:                   subnets,
-	}
-	workflowOptions := client.StartWorkflowOptions{
-		TaskList:                     "pipeline",
-		ExecutionStartToCloseTimeout: 40 * time.Minute, // TODO: lower timeout
+	reqNodePools := createNodePoolsFromPKERequest(request.PKE.NodePools)
+	reqNodePoolsMap := map[string]pkeworkflow.NodePool{}
+	for _, np := range reqNodePools {
+		reqNodePoolsMap[np.Name] = np
 	}
-	exec, err := workflowClient.ExecuteWorkflow(ctx, workflowOptions, pkeworkflow.UpdateClusterWorkflowName, input)
-	if err != nil {
-		return err
+
+	clusterNodePools := createNodePoolsFromPKENodePools(c.GetNodePools())
+	clusterNodePoolsMap := map[string]pkeworkflow.NodePool{}
+	for _, np := range clusterNodePools {
+		clusterNodePoolsMap[np.Name] = np
 	}
 
-	err = c.SetCurrentWorkflowID(exec.GetID())
-	if err != nil {
-		return err
+	var nodePoolsToAdd []pkeworkflow.NodePool
+	var nodePoolsToUpdate []pkeworkflow.NodePool
+	var nodePoolsToDelete []pkeworkflow.NodePool
+
+	for _, np := range clusterNodePools {
+		if np.Master || np.Name == "master" {
```

Comment:
Are there any known usage where `np.Master` _is_ true and `np.Name` _is not_ master?
If not, every initialisation of `PKENodePool` should set `np.Master = true` if `np.Name == "master"`. Handling `if np.Master || np.Name == "master" {` like everywhere in code is error prone.